tests/integration/states/test_file.py

```diff
@@ -573,6 +573,36 @@ def test_managed_contents(self):
                 if os.path.exists(managed_files[typ]):
                     os.remove(managed_files[typ])
 
+    def test_managed_contents_with_contents_newline(self):
+        '''
+        test file.managed with contents by using the default content_newline
+        flag.
+        '''
+        contents = r'test_managed_contents_with_newline_one'
+        name = os.path.join(TMP, 'foo')
+
+        # Create a file named foo with contents as above but with a \n at EOF
+        self.run_state('file.managed', name=name, contents=contents,
+                       contents_newline=True)
+        with salt.utils.files.fopen(name, 'r') as fp_:
```

Comment:
Wouldn't it be appropriate to open this in `rb` mode instead? 🤔

Reply:
Hmm, you aren't wrong! I must have forgotten the `b`

Reply:
Actually no I believe it should remain as r. I thought about this and the PR was incorrectly named (definitely my fault) but opening the file in binary mode and then reading it will produce different results based on the platform, i.e. Window's will read the contents differently than a linux platform. So I decided to just open the file in r as it's more in-line with the change in logic itself. Does that make sense?